scim2-sdk-common/src/main/java/com/unboundid/scim2/common/messages/PatchOperation.java

```diff
@@ -54,11 +54,11 @@
     property = "op")
 @JsonSubTypes({
     @JsonSubTypes.Type(value = PatchOperation.AddOperation.class,
-        name="add"),
+        name="add", names= {"Add", "ADD"}),
     @JsonSubTypes.Type(value = PatchOperation.RemoveOperation.class,
-        name="remove"),
+        name="remove", names= {"Remove", "REMOVE"}),
     @JsonSubTypes.Type(value = PatchOperation.ReplaceOperation.class,
-        name="replace")})
+        name="replace", names= {"Replace", "REPLACE"})})
```

Comment:
The [Javadoc](https://javadoc.io/static/com.fasterxml.jackson.core/jackson-annotations/2.14.2/index.html?com/fasterxml/jackson/annotation/JsonSubTypes.html) isn't super clear about how the `name` and `names` annotation elements interact with each other.

The documented description for the `name` element is:

> Logical type name used as the type identifier for the class, if defined; empty String means "not defined". Used unless names is defined as non-empty.

And the documented description for the `names` element is:

> optional) Logical type names used as the type identifier for the class: used if more than one type name should be associated with the same type.

My reading of this is that the two elements are mutually exclusive. The "Used unless names is defined as non-empty" portion of the `name` description suggests that the `name` element is ignored if `names` is defined and non-empty. That's also backed up by the "used if more than one type name should be associated with the same type" portion of the `names` description.

Given that, it sounds like if we're going to use the `names` element, then we want its value to include the complete set of names that should be allowed. In that case, I think the safe thing to do would be to have

`name="add", names={"add", "Add", "ADD"}`

rather than

`name="add", names={"Add", "ADD"}`

And have similar changes for the other operation types.

And if you haven't already, I'd recommend testing your changes against other types of SCIM implementations and not just Azure AD.

Reply:
I'll make sure to check this against other implementations. As for the other operation types, PATCH is the only operation utilizing this JsonSubTypes interface.

Reply:
Sorry. For "other operation types", I meant the add, remove, and replace operation types within the patch request, although it's also good to know that it's not used for other types of SCIM requests.